GiniVision/Classes/Core/Screens/Help/HelpMenuViewController.swift

```diff
@@ -135,9 +174,24 @@ extension HelpMenuViewController {
     }
     
     override public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        if let viewController = viewController(forRowWithId: items[indexPath.row].id) {
-            navigationController?.pushViewController(viewController, animated: true)
+        let item = items[indexPath.row]
+        let viewController = item.viewController
+        viewController.setupNavigationItem(usingResources: backToMenuButtonResource,
+                                           selector: #selector(back),
+                                           position: .left,
+                                           target: self)
+        
+        if let imageNoResultsViewController = viewController as? ImageAnalysisNoResultsViewController {
+            imageNoResultsViewController.didTapBottomButton = {
```

Comment:
Here, you ruin the "magic" of encapsulating the type of view controller. Can you find a way to setup the `ImageAnalysisNoResultsViewController` in a way that it won't break the abstraction?

Reply:
Can't make it with the new `HelpMenuViewControllerDelegate`, but since it's marked as deprecated internally, it shouldn't be a problem anymore.

Reply:
When? it's going to take a long time until we actually remove it.